src/pykoala/corrections/wavelength.py

```diff
@@ -102,6 +107,240 @@ def apply(self, rss):
+
+        # Detect absorption features
+        median_continuum_ratio = np.nanmedian(solar_spectra / solar_continuum)
+        weights = np.abs(solar_spectra / solar_continuum -  median_continuum_ratio)
+        return weights
+
+    def compute_grid_of_models(self, pix_shift_array, pix_std_array, pix_array,
+                              sun_intensity, weights):
+        print("Computing grid of models") #TODO
+        models_grid = np.zeros((pix_shift_array.size, pix_std_array.size,
+                           sun_intensity.size))
+        weights_grid = np.zeros((pix_shift_array.size, pix_std_array.size,
+                           sun_intensity.size))
+        for i, velshift in enumerate(pix_shift_array):
+                for j, gauss_std in enumerate(pix_std_array):
+                    new_pixel_array = pix_array + velshift
+                    
+                    interp_sun_intensity = flux_conserving_interpolation(
+                        new_pixel_array, pix_array, sun_intensity)
+                    interp_sun_intensity = gaussian_filter(
+                        interp_sun_intensity, gauss_std)
+                    models_grid[i, j] = interp_sun_intensity
+
+                    interp_sun_weight = flux_conserving_interpolation(
+                        new_pixel_array, pix_array, weights)
+                    interp_sun_weight = gaussian_filter(
+                        interp_sun_weight, gauss_std, truncate=2.0)
+                    interp_sun_weight /= np.nansum(interp_sun_weight)
+                    weights_grid[i, j] = interp_sun_weight
+        return models_grid, weights_grid
+
+    def compute_shift_from_twilight(self, data_container, logspace=True,
+                                    sun_window_size_aa=20, keep_features_frac=0.1,
+                                    response_window_size_aa=200,
+                                    wave_range=None,
+                                    pix_shift_array=np.arange(-5, 5, 0.1),
+                                    pix_std_array=np.arange(0.1, 3, 0.1),
+                                    inspect_fibres=None):
+        if logspace:
+            new_wavelength = np.geomspace(data_container.wavelength[0],
+                                          data_container.wavelength[-1],
+                                          data_container.wavelength.size)
+            rss_intensity = np.array([flux_conserving_interpolation(
+                new_wavelength, data_container.wavelength, fibre
+                ) for fibre in data_container.rss_intensity])
+        else:
+            new_wavelength = data_container.wavelength
+            rss_intensity = data_container.rss_intensity
+        
+        # Interpolate the solar spectrum to the new grid of wavelengths
+        sun_intensity = flux_conserving_interpolation(
+        new_wavelength, self.sun_wavelength, self.sun_intensity)
+
+        # Make an array of weights to focus on the absorption lines
```

Comment:
What about sky emission lines?

Reply:
There are no sky lines in the twilight spectra, the problematic ones are the tellurics